lib/site_settings/validations.rb

```diff
@@ -189,7 +189,9 @@ def validate_backup_location(new_val)
       validate_error(:s3_backup_requires_s3_settings, setting_name: "s3_backup_bucket")
     end
 
-    unless SiteSetting.s3_use_iam_profile
+    # Only validate credentials if user is providing them
```

Comment:
we should be careful about saying e.g. "user" and this should say "admin" (or whatever we use elsewhere)